pynicotine/search.py

```diff
@@ -289,6 +340,50 @@ def do_wishlist_search_interval(self):
                 self.do_wishlist_search(search.token, term)
                 break
 
+    def _add_results(self, search, results, user, country_code, queue, speed, has_free_slots, is_private=False):
+
+        for _code, fullpath, size, _ext, attributes in results:
+            if search.num_results >= search.max_results:
+                break
+
+            fullpath_lower = fullpath.lower()
+
+            if any(word in fullpath_lower for word in search.term_words_ignore):
+                # Filter out results with filtered words (e.g. nicotine -music)
+                log.add_debug(("Filtered out excluded search result %(filepath)s from user %(user)s for "
+                               'search term "%(query)s"'), {
+                    "filepath": fullpath,
+                    "user": user,
+                    "query": search.term
+                })
+                continue
+
+            if not any(word in fullpath_lower for word in search.term_words_include):
+                # Certain users may send us wrong results, filter out such ones
+                log.add_search(_("Filtered out incorrect search result %(filepath)s from user %(user)s for "
+                                 'search query "%(query)s"'), {
+                    "filepath": fullpath,
+                    "user": user,
+                    "query": search.term
+                })
+                continue
+
+            search.num_results += 1
+            search.results[user].append(
+                SearchResult(
+                    num=search.num_results,
+                    user=user,
+                    file_path=fullpath,
```

Comment:
`file_path` -> `fullpath` (or unify all to `full_path`)

Also consider adding `folder`** and `filename` into the `SearchResult()` would help to reduce the amount of rsplit() and join() operations elsewhere.

** _Maybe `dir` or `directory` is appropriate for references to remote folders._

Reply:
The `path` term is too ambiguous as to a folder without a filename.

Consider adding `folder`** and `filename` into the `SearchResult()` would help to reduce the amount of rsplit() and join() operations elsewhere.

** or maybe `dir` or `directory` would be more appropriate for references to remotely indexed folders.

Reply:
Disregard this review, I agree that file_path makes the most sense.